### ISSUE TITLE
Flutter app running in iOS Simulator drains battery quickly

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#11452]
Unclear if this is Flutter's fault or iOS Simulator?  This was reported by @harveynick.

**Comment by Hixie** [comment]
Which battery?

**Comment by eseidelGoogle** [comment]
The Mac's battery. :)

**Comment by PerrchicK** [comment]
Occurs to me as well, every time I use with Flutter. I can see it clearly in the energy impact.

I first noticed this reaction when playing a flare animation. But later I noticed it appears in every render of the UI. Am I using it wrong? Does everybody see this in their projects?

I can accept the average usage meter, since it’s an average there’s a good explanation for that. But why does it affect the battery so much? I want to build an app for production usage but doesn't seem quite healthy.

Attaching a video:
https://www.dropbox.com/s/ftxn32sk9xypkto/flutter-energy-impact.mp4?dl=0
And my code as well (Flutter, iOS & Android), in case you'll find some mistake there:
https://www.dropbox.com/s/zay4c7iurxvoe0u/battery_impact_example.zip?dl=0

Thanks a lot!

**Comment by PerrchicK** [comment]
This case caught my attention regarding to the GPU:
https://www.dropbox.com/s/ozdwu3t42jl4d2h/flutter-animation-energy-impact.mp4?dl=0
(Running on a real iPhone Xs device, it supposed to have a strong battery)

I wanted to learn how to play an animation made from flare (https://proandroiddev.com/how-to-flare-a-flutter-app-part-2-flutter-integration-b188b0661d17) but then saw the GPU usage and battery impact.

I'll be more than glad to know that I shouldn't be worried, what do you think? Is that an issue that needs to be fixed? When?

Or is this a normal behaviour?

Thank you :)

**Comment by mrgoonie** [comment]
I've got the same issue with my app, the battery was drained a lot while using. I try to test a screen with and without animation, there is a different, so I think somehow I need to avoid doing much animating.. but, that's not fun anymore, animation is a need for better user experience, I hope Flutter team look on this issue seriously 😄

**Comment by eseidelGoogle** [comment]
This bug isn't particularly actionable, as written.

I would love for us to take action here, but a specific example would be helpful.  Does someone have an app in mind (ideally open source) which they're finding is draining battery too quickly?

**Comment by jonahwilliams** [comment]
Does a Flutter app actually draw more power than an example UIkit app?

**Comment by PerrchicK** [comment]
Is this example helps?
https://www.dropbox.com/s/dt32jpuihcg7gm4/battery_impact_example.zip?dl=0

@eseidelGoogle, would you like me to open a new issue that explains my use case better?

Thanks, I appreciate your reply.

**Comment by eseidelGoogle** [comment]
@PerrchicK thanks for the example.  Yes, if you're willing to file a new issue I think that's best.

I think we can also close this one once that's done, I don't think this issue is very useful anymore, but new issues (ideally with examples) would be very useful!  Thank you!

**Comment by PerrchicK** [comment]
Done, thank you.
reference: https://github.com/flutter/flutter/issues/26895

**Comment by zoechi** [comment]
Somewhat related #13203, #6135

**Comment by arjunhadke7** [comment]
> https://www.youtube.com/watch?v=I_VacxqDKDg

Hi, this link appears to show a page with Video Unavailable. Can you share a working link again, please?

**Comment by PerrchicK** [comment]
Sorry for my really late response, please ignore all my previous comments. I test it with wrong conditions (I was profiling on DEBUG mode on Xcode 🤦‍♂️), that's why I removed the YouTube videos link I share before.

I'm actually very satisfied by Flutter and its battery impact today.
I would close this issue if I could.

**Comment by arjunhadke7** [comment]
> https://www.youtube.com/watch?v=I_VacxqDKDg

The battery drain issue is consistent with any older mac. I think that does not impact a newer machine. This, probably is laptop battery/hardware issue caused as the laptop is older and the battery has gone through numerous cycles rather than a software issue. 

Are other people seeing this issue even on a newer mac after fully updating/upgrading Flutter SDK?

**Comment by benjamincombes** [comment]
I don't think it's really related to the battery itself. A real-life example from today:
I took the train with 100% battery on my laptop (a 2019 MacBook Pro, not that old). After working for 1h15 on Android Studio, the battery was at 86%.
At that point, I started launching my app (quite basic, and 95% of the time, I was not using it, just coding) on the iOS simulator. After about 10-15 minutes, my battery was at 26%...!
Then I stopped the Flutter App & simulator, and after working 1 more hour, I had a 5% battery left.
There really is a huge battery drain on Flutter use on the iOS simulator (and I can definitely can say it, as all fans are always on as soon as the Flutter app starts, even when doing nearly nothing with it).
